src/pages/smart-contracts/clarity-values-guide.md

```diff
@@ -0,0 +1,143 @@
+
+Here are examples of how to construct each type of Clarity value:
+
+### Booleans
+```javascript
+const t = trueCV();
+const f = falseCV();
+```
+### Optional Values
+```javascript
+const nothing = noneCV();
+const something = someCV(t);
+```
+
+### Buffers
+```javascript
+const buffer = Buffer.from('foo');
+const bufCV = bufferCV(buffer);
+```
+
+### Integers
+```javascript
+const i = intCV(-10);
+const u = uintCV(10);
+```
+
+### Strings
+```javascript
+const ascii = stringAsciiCV('hello world');
+const utf8 = stringUtf8CV('hello 🌾');
+```
+
+### Principals
+```javascript
+const address = 'SP2JXKMSH007NPYAQHKJPQMAQYAD90NQGTVJVQ02B';
+const contractName = 'contract-name';
+
+const spCV = standardPrincipalCV(address);
+const cpCV = contractPrincipalCV(address, contractName);
+```
+
+### Response Values
+```javascript
+const errCV = responseErrorCV(trueCV());
+const okCV = responseOkCV(falseCV());
+```
+
+### Tuples
+```javascript
+const tupCV = tupleCV({
+  'a': intCV(1),
+  'b': trueCV(),
+  'c': falseCV()
+})
+```
+
+### Lists
+```javascript
+const l = listCV([trueCV(), falseCV()])
+```
+
+## Using Clarity Values
+Now that you know how to construct Clarity values, you can use them to build `contract-call` transactions that call smart contract functions.
+
+Here's an example of how this is done:
+
+```javascript
+import {
+  makeContractCall,
+  uintCV,
+  stringAsciiCV,
+  StacksMainnet,
+  broadcastTransaction
+} from '@stacks/transactions';
+
+const network = new StacksMainnet();
+
+const txOptions = {
+  contractAddress: 'SPBMRFRPPGCDE3F384WCJPK8PQJGZ8K9QKK7F59X',
+  contractName: 'contract_name',
+  functionName: 'contract_function',
+  functionArgs: [uintCV(10), stringAsciiCV('test')],
+  senderKey: 'b244296d5907de9864c0b0d51f98a13c52890be0404e83f273144cd5b9960eed01',
+  validateWithAbi: true,
+  network,
+};
+
+const transaction = await makeContractCall(txOptions);
+
+broadcastTransaction(transaction, network);
+```
+
+## Utilizing Clarity Values from Transaction Responses
+If you successfully submit a `contract-call` transaction that calls a Clarity function, you may receive a resulting Clarity value in response.
+
+If using the transaction library's `callReadOnlyFunction()` method, the result will be a `ClarityValue`. As mentioned above, `ClarityValues` are simply javascript objects containing a value and its associated Clarity type information. These object types are defined [here](https://github.com/blockstack/stacks.js/tree/1f2b5fd8bdf1c2b5866e8171163594d7708a8c7a/packages/transactions/src/clarity/types).
+
+### Deserializing Clarity Values from Hex
```

Comment:
can we also talk about `deserializeCV` and `serializeCV`?

Reply:
Are these functions really important enough to be mentioned in this guide? When would a dev need to use them?

My understanding is that developers will mainly be serializing/deserializing entire transactions, not individual Clarity Values. Is this incorrect?

Reply:
I think contract calls from JS would need to use these methods, see the [stacking tutorial](https://docs.blockstack.org/stacks-blockchain/integrate-stacking#step-4-verify-stacking-eligibility)